src/components/markdown-renderer/basic-markdown-renderer.tsx

```diff
@@ -36,53 +25,25 @@ export const BasicMarkdownRenderer: React.FC<BasicMarkdownRendererProps & Additi
   content,
   wide,
   componentReplacers,
-  onConfigureMarkdownIt,
+  markdownIt,
   documentReference,
   onBeforeRendering
 }) => {
   const maxLength = useSelector((state: ApplicationState) => state.config.maxDocumentLength)
 
-  const markdownIt = useMemo(() => {
-    const md = new MarkdownIt('default', {
-      html: true,
-      breaks: true,
-      langPrefix: '',
-      typographer: true
-    })
-
-    md.use(emoji, {
-      defs: combinedEmojiData
-    })
-    md.use(abbreviation)
-    md.use(definitionList)
-    md.use(subscript)
-    md.use(superscript)
-    md.use(inserted)
-    md.use(marked)
-    md.use(footnote)
-    md.use(imsize)
-
-    if (onConfigureMarkdownIt) {
-      onConfigureMarkdownIt(md)
-    }
-
-    md.use(linkifyExtra)
-    if (process.env.NODE_ENV !== 'production') {
-      md.use(MarkdownItParserDebugger)
-    }
-
-    return md
-  }, [onConfigureMarkdownIt])
-
   const oldMarkdownLineKeys = useRef<LineKeys[]>()
   const lastUsedLineId = useRef<number>(0)
 
   const markdownReactDom: ReactElement[] = useMemo(() => {
     if (onBeforeRendering) {
       onBeforeRendering()
     }
+    let md: MarkdownIt = basicMarkdownRender()
+    if (markdownIt) {
+      md = markdownIt
+    }
```

Comment:
Why are you creating a markdown renderer in every case? IMHO the markdown it prop shouldn't be optional.